pkgs/top-level/all-packages.nix

```diff
@@ -11869,6 +11870,7 @@ in
     # Can only do this is in the native case, otherwise we might get infinite
     # recursion if `targetPackages.stdenv.cc.cc` itself uses `gccForLibs`.
       then targetPackages.stdenv.cc.cc
+    else if targetPlatform.isiOS then null
```

Comment:
Instead, could you add this condition to https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/cc-wrapper/default.nix#L64. This way we can keep the android / ios stuff in one place.

Reply:
Done. If there’s a way to fix gcc compilation instead, that might be preferred - it looks like “real” libs (eg eigen and cxxopts) have some issues being built with the current toolchain (missing header files) after I’ve gotten past these errors. I don’t think a lack of gcc is the issue there, but I don’t 100% understand how building works, so I can’t rule it out.

To be clear, they initially have the same errors as hello, but whereas these changes seem to be sufficient for building hello, more may be needed for libraries which need to link libmath or include <iostream>.

Reply:
So gccForLibs is just the gcc c++ and other runtime libraries. You shouldn’t need these on iOS where libc++ is provided.